File: backend/buylogic/src/main/java/com/buylogic/model/PurchaseOrderItem.java
package com.buylogic.model;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "purchase_order_item")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PurchaseOrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idPurchaseOrderItem;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_purchase_order", nullable = false)
    private PurchaseOrder purchaseOrder;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_product", nullable = false)
    private Product product;

    @Column(name = "quantity_ordered", nullable = false, precision = 14, scale = 3)
    private BigDecimal quantityOrdered;

    @Column(name = "quantity_received", nullable = false, precision = 14, scale = 3)
    private BigDecimal quantityReceived = BigDecimal.ZERO;

    @Column(name = "unit_price", nullable = false, precision = 12, scale = 2)
    private BigDecimal unitPrice;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend/src/types/purchaseOrderItem.ts
export interface PurchaseOrderItem {
  idPurchaseOrderItem: number;
  idPurchaseOrder: number;
  idProduct: number;

  productReference?: string | null;
  productName?: string | null;

  quantityOrdered: number;
  quantityReceived: number;
  unitPrice: number;
}

export interface PurchaseOrderItemCreate {
  idPurchaseOrder: number;
  idProduct: number;
  quantityOrdered: number;
  quantityReceived?: number;
  unitPrice: number;
}

export interface PurchaseOrderItemUpdate {
  quantityOrdered?: number;
  quantityReceived?: number;
  unitPrice?: number;
}